src/node/channels.h

```diff
@@ -246,93 +258,346 @@ namespace ccf
-      bool complete, const uint8_t* data, size_t size)
+    std::vector<uint8_t> get_signed_key_share(bool with_salt)
     {
-      if (status == ESTABLISHED)
+      return sign_key_share(kex_ctx.get_own_key_share(), with_salt);
+    }
+
+    CBuffer extract_buffer(const uint8_t*& data, size_t& size) const
+    {
+      if (size == 0)
       {
-        return false;
+        return {};
+      }
+
+      auto sz = serialized::read<size_t>(data, size);
+      CBuffer r(data, sz);
+
+      if (r.n > size)
+      {
+        LOG_FAIL_FMT(
+          "Buffer header wants {} bytes, but only {} remain", r.n, size);
+        r.n = 0;
+      }
+      else
+      {
+        data += r.n;
+        size -= r.n;
+      }
+
+      return r;
+    }
+
+    bool verify_peer_certificate(CBuffer pc)
+    {
+      if (pc.n != 0)
+      {
+        peer_cert = crypto::Pem(pc);
+        peer_cv = crypto::make_verifier(peer_cert);
+
+        if (!peer_cv->verify_certificate({&network_cert}))
+        {
+          LOG_FAIL_FMT("Peer certificate verification failed");
+          reset();
+          return false;
+        }
+
+        LOG_DEBUG_FMT(
+          "New peer certificate: {}\n{}",
+          peer_cv->serial_number(),
+          peer_cert.str());
       }
 
-      auto network_pubk = crypto::make_public_key(network_kp->public_key_pem());
+      return true;
+    }
 
-      auto peer_public_size = serialized::read<size_t>(data, size);
-      auto peer_public_start = data;
+    bool verify_peer_signature(CBuffer msg, CBuffer sig)
+    {
+      LOG_DEBUG_FMT(
+        "Verifying peer signature with peer certificate serial {}",
+        peer_cv ? peer_cv->serial_number() : "no peer_cv!");
 
-      if (peer_public_size > size)
+      if (!peer_cv || !peer_cv->verify(msg, sig))
       {
         LOG_FAIL_FMT(
-          "Peer public key header wants {} bytes, but only {} remain",
-          peer_public_size,
-          size);
+          "Node channel peer signature verification failed for {} with "
+          "certificate serial {}",
+          peer_id,
+          peer_cv->serial_number());
         return false;
       }
 
-      data += peer_public_size;
-      size -= peer_public_size;
+      return true;
+    }
+
+    // Protocol overview:
+    //
+    // initiate()
+    // > key_exchange_init message
+    // consume_initiator_key_share() [by responder]
+    // < key_exchange_response message
+    // consume_responder_key_share() [by initiator]
+    // > key_exchange_final message
+    // check_peer_key_share_signature() [by responder]
+    // both reach status == ESTABLISHED
+
+    bool consume_responder_key_share(const std::vector<uint8_t>& data)
+    {
+      return consume_responder_key_share(data.data(), data.size());
+    }
 
-      auto signature_size = serialized::read<size_t>(data, size);
-      auto signature_start = data;
+    bool consume_responder_key_share(const uint8_t* data, size_t size)
+    {
+      LOG_DEBUG_FMT("status == {}", status);
```

Comment:
Can we make this, and all other log messages in this file (except the channel initiation and establishment) `LOG_TRACE_FMT`?